### PR TITLE
Add creation function for Division A field

### LINKED ISSUE #1400: Replace TestUtil::createSSLDivBField with equivalent static Field constructor
### Description of the task

The `createSSLDivBField` helper function makes more sense as a static `Field` constructor so test utils don't have to be pulled in everywhere. This function should be moved to `Field` and all usaages replaced.

### Acceptance criteria

- [ ] as titled

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

Moved the `::TestUtil::createSSLDivBField()` function to the `Field` as a static constructor. Doing this now because I realized I need it there in order to use in the simulated test library / text fixture target (because it's not a `cc_test` and so I can't depend on `test_util` because it's marked as `test_only=True`)

### Testing Done

Added more checks to the creation unit test. Now all "getter" functions of the field are covered.

### Resolved Issues

resolves #1400 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by jonathanlew — COMMENTED** [review]
Can we quickly add div A field too?

**Review by garethellis0 — CHANGES_REQUESTED** [review]
I'd really like to avoid this change if possible, since the specific dimensions of the SSL Div B field don't seems like they belong in the generic field class. Perhaps let's mark the test fixture stuff as `test_only` as well? That _should_ work.

**Comment by MathewMacDougall** [comment]
> I'd really like to avoid this change if possible, since the specific dimensions of the SSL Div B field don't seems like they belong in the generic field class.

I don't see why not? `Angle` has constructors for `zero`, `half`, etc. angles which is similar (not 100% same comparison).

**Edit:** Since we currently don't have any use-cases outside of tests I'll leave it where it is.


> Perhaps let's mark the test fixture stuff as test_only as well? That should work.

True I also completely did not think of that.

**Comment by MathewMacDougall** [comment]
Ok I left the `DivB` field constructor back in `TestUtil` but improved the unit test. I also added a function and test for creating a Division A field @jonathanlew

File: src/software/test_util/test_util_test.cpp [hunk 1]
```diff
@@ -7,37 +7,126 @@
...
+
+    EXPECT_EQ(Point(-6, 0.9), field.friendlyGoalpostPos());
+    EXPECT_EQ(Point(-6, -0.9), field.friendlyGoalpostNeg());
+    EXPECT_EQ(Point(6, 0.9), field.enemyGoalpostPos());
+    EXPECT_EQ(Point(6, -0.9), field.enemyGoalpostNeg());
+
+    EXPECT_DOUBLE_EQ(0.3, field.boundaryMargin());
 }
 
+
```
Comment (jonathanlew) [inline src/software/test_util/test_util_test.cpp]: nit extra new line

Comment (MathewMacDougall) [reply src/software/test_util/test_util_test.cpp]: removed

**Review by jonathanlew — APPROVED** [review]
LGTM! :field_hockey:

**Review by kelvinkoon — APPROVED** [review]

**Review by akhilveeraghanta — APPROVED** [review]
⚽

**Review by garethellis0 — APPROVED** [review]

**Pull request merged**